web/src/drafts_overlay_ui.ts

```diff
@@ -126,128 +187,165 @@ export function handle_keyboard_events(event_key: string): void {
+            return !is_checkbox_icon_checked($(this));
+        });
+        const check_boxes = $unchecked_checkboxes.length > 0;
+        $(".draft-selection-checkbox").each(function () {
+            toggle_checkbox_icon_state($(this), check_boxes);
         });
+        update_bulk_delete_ui();
+    });
 
-        $(".select-drafts-button").on("click", (e) => {
-            e.preventDefault();
-            const $unchecked_checkboxes = $(".draft-selection-checkbox").filter(function () {
-                return !is_checkbox_icon_checked($(this));
-            });
-            const check_boxes = $unchecked_checkboxes.length > 0;
-            $(".draft-selection-checkbox").each(function () {
-                toggle_checkbox_icon_state($(this), check_boxes);
+    $(".delete-selected-drafts-button").on("click", () => {
+        $(".drafts-list")
+            .find(".draft-selection-checkbox.fa-check-square")
+            .closest(".overlay-message-row")
+            .each(function () {
+                remove_draft($(this));
             });
-            update_bulk_delete_ui();
-        });
+        update_bulk_delete_ui();
+    });
+}
+
+function rerender_drafts(): void {
+    const all_drafts = drafts.draft_model.get();
+    const narrow_drafts_raw = drafts.filter_drafts_by_compose_box_and_recipient(all_drafts);
+    const other_drafts_raw = _.pick(
+        all_drafts,
+        _.difference(Object.keys(all_drafts), Object.keys(narrow_drafts_raw)),
+    );
+    const narrow_drafts = format_drafts(narrow_drafts_raw);
+    const other_drafts = format_drafts(other_drafts_raw);
+    const narrow_drafts_header = get_header_for_narrow_drafts();
+
+    const rendered = render_drafts_list({
+        narrow_drafts_header,
+        narrow_drafts,
+        other_drafts,
+    });
+    const $drafts_container = $("#drafts_list_container");
+    $drafts_container.empty().append($(rendered));
 
-        $(".delete-selected-drafts-button").on("click", () => {
-            $(".drafts-list")
-                .find(".draft-selection-checkbox.fa-check-square")
-                .closest(".overlay-message-row")
-                .each(function () {
-                    remove_draft($(this));
-                });
-            update_bulk_delete_ui();
+    const $drafts_table = $("#drafts_table");
+    if ($drafts_table.find(".overlay-message-row").length > 0) {
+        $("#drafts_table .no-drafts").hide();
+
+        const $rendered_drafts = $drafts_table.find(
+            ".message_content.rendered_markdown.restore-overlay-message",
+        );
+        $rendered_drafts.each(function () {
+            rendered_markdown.update_elements($(this));
         });
+
+        $(".select-drafts-button").show();
+        $(".delete-selected-drafts-button").show();
     }
+    update_rendered_drafts(narrow_drafts.length > 0, other_drafts.length > 0);
+    update_bulk_delete_ui();
+    setup_event_handlers();
+}
```

Comment:
This should be refactored to share more code with `launch`, rather than being a near-duplicate of it.

Near-duplicate functions like this almost inevitable will diverge from the main code path in uintended and buggy ways.